tests/pytests/unit/spm/test_spm.py

```diff
@@ -0,0 +1,213 @@
+import os
+import shutil
+
+import pytest
+import salt.config
+import salt.spm
+import salt.utils.files
+from tests.support.mixins import AdaptedConfigurationTestCaseMixin
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture()
+def f1_content():
+    _F1 = {
+        "definition": {
+            "name": "formula1",
+            "version": "1.2",
+            "release": "2",
+            "summary": "test",
+            "description": "testing, nothing to see here",
+        }
+    }
+
+    _F1["contents"] = (
+        (
+            "FORMULA",
+            (
+                "name: {name}\n"
+                "version: {version}\n"
+                "release: {release}\n"
+                "summary: {summary}\n"
+                "description: {description}"
+            ).format(**_F1["definition"]),
+        ),
+        ("modules/mod1.py", "# mod1.py"),
+        ("modules/mod2.py", "# mod2.py"),
+        ("states/state1.sls", "# state1.sls"),
+        ("states/state2.sls", "# state2.sls"),
+    )
+    return _F1
+
+
+@pytest.mark.destructive_test
```

Comment:
Why is this test destructive?
Creating/deleting files in the test suite temp directories is not considered a destructive operation, it's expected.